### ISSUE TITLE
General questions

### ISSUE THREAD

**Issue by bbczeuz** [rbalykov/dmx-multiverse#4]
What are the benefits of an updated DMX standard to already established open industry standards like:

Artnet
- UDP Unicast/Broadcast based
- Multiverse
- (somewhat limited) Device discovery

RDM
- RS485 (Halfduplex) based
- No data transmission
- (somewhat limited) Device discovery

ACN
- UDP Unicast/Multicast/Broadcast based
- Neat design using Multicast groups for data distribution
- Multiverse
- Powerful device discovery
IMHO, the ACN standard is too flexible or powerful to be implemented in full in many (most?) scenarios, but I really like the approach. 
AFAIR ACN only proposes to use UDP over Ethernet, but can be transported using other means.
How about working on this instead of proposing a new standard (xkcd...)?

**Comment by rbalykov** [comment]
For TCP/IP-based topology, there's nothing to change.
Changes are over XLR-Socket-Terminated microphone-type cable.

1) While it can supply 10 Mbit over the stage (WS2812 LEDs need 1 Mbit, as instance), it takes just 250 Kbit to carry.
2) Frame size is limited to 512 bytes. Far in 1989, that was enough to drive tungsten-lamp dimmers, but not now-day's multimedia.
3) Framerate done ugly - using full frame, it gives 44 fps, hardly bound to TV's 50/60.


Art-net/ACN - no changes, hardware just loads multiple universes to XLR cable.
RDM - no changes, your GrandMA console is still good.  Fixtures do the same as before.

**Comment by vanvught** [comment]
> (WS2812 LEDs need 1 Mbit, as instance)

Not really. 

- The FPS limitation is with the length of your strip. With a physical DMX universe, you can control 170 RGB led's at 44FPS.  
- The limitation with the 512 slots in physical DMX is that you can control a maximum of 170 RGB led's or 128 RGBW led's.

**Comment by vanvught** [comment]
>  Changes are over XLR-Socket-Terminated microphone-type cable.

Microphone-type cables cannot be used for DMX-512. 

A reference: [https://tmb.com/docs/tech-tips/TT8-MicCableforDMX.pdf](https://tmb.com/docs/tech-tips/TT8-MicCableforDMX.pdf)

**Comment by rbalykov** [comment]
> Not really.

Using 800 kbaud, it uses it's famous "unique" bitcoding that makes things look like 1000 Kbaud

> The FPS limitation is with the length of your strip. With a physical DMX universe, you can control 170 RGB led's at 44FPS.

Thus is's still possible to drive 30 fps, not the 50/60. To do so, MAB/IDLE have to be tweaked in some vendor-specific fashion, undefined in protocol.

> The limitation with the 512 slots in physical DMX is that you can control a maximum of 170 RGB led's or 128 RGBW led's.

So, *ClayPaky B-EYE K20* fixture utilises 37 RGBW LEDs, that takes 148 slots to drive them as *pixelmapping*.
Adding 4th fixture like this to stage setup needs one more universe, one more XLR port, one more 20-meter XLR cable.

> Microphone-type cables cannot be used for DMX-512.

Cables designed to carry audio are not allowed (or recommended) in here, but widely used, and they perform that job. Specific 120-Ohm cable in fact designed the same way - it's shielded twisted pair. So that's called as *microphone-type*.

**Comment by vanvught** [comment]
> Using 800 kbaud, it uses it's famous "unique" bitcoding that makes things look like 1000 Kbaud

The bit coding is simple RTZ. Very easy to implement -> https://www.gd32-dmx.org/pixel.html#section1
Not sure about ` look like 1000 Kbaud`
This send data at speeds of 800Kbps (with the Reset time) is the general limitation for your FPS depending on the strip length.

>  MAB/IDLE have to be tweaked in some vendor-specific fashion, undefined in protocol.

Reference `E1.11 – 2008 (R2013)` When you have less than 512 slots in use, then you can do a lot with FPS within the protocol specification.
![Schermafbeelding](https://user-images.githubusercontent.com/4319913/184371400-2e00f049-4757-47e9-a12b-05a71e9f8a8b.png)

**Comment by rbalykov** [comment]
> When you have less than 512 slots in use.

Yep. Less slots = more cables.
More consoles, more splitters, more time to set it up, etc.

**Comment by kripton** [comment]
I agree that DMX-512-A is quite dated and with current RGB-LED-Fixtures taking up lots of channels, DMX-512-A reaches practical limits quite fast.
But DMX is still used to drive fixtures, in the end. So even if there was a new standard to control lighting fixtures (and similar), it would require adopters. And I would rather see professional equipment to directly connect via Ethernet -> sACN/ArtNet than inventing another, new standard using existing 3-, 5, or CAT-cable.
I really appreciate your thoughts and the time you invested but I don't see that it would see a big adoption or wide use.

If it was about re-using existing cabling, I would rather see either Two-Wire-Ethernet or some DSL-approach or G.HN standard and then using existing network-based protocols (sACN, ArtNet) on top of those.

**Comment by rbalykov** [comment]
> But DMX is still used to drive fixtures, in the end. So even if there was a new standard to control lighting fixtures (and similar), it would require adopters. 

So, 250k rated dmx-512/rdm stays untouched.
No changes to fixtures or consoles. 1989 standard is easily inherited untouched.

> If it was about re-using existing cabling

Reusing not a Cat5/6/Fiber cable, but XLR-terminated cable to drive megabit to ws2812, pixelmapping and so on.
8 megabit as well, in fact. 

So you can put on same cable:

1 universe, if your splitter has 250 kbit-limited 485 chip.
8 universes, if chip limited to 2.5 mbit.
32 universes, if limited to 10 mbit.

But it needs old DMX splitter to be re-sertified to be stable at higher rates.

**Comment by rbalykov** [comment]
> So you can put on same cable:
> 
> 1 universe, if your splitter has 250 kbit-limited 485 chip.
> 8 universes, if chip limited to 2.5 mbit.
> 32 universes, if limited to 10 mbit.

Starcode:
SC 00 untouched
Exra SCs to address new frames

Splitters: 
Nothing to do if yours drives 10 mbit.
Solder new 485 ICs if you want your 2.5M to be 10M or just not touch it and use it at 2M.

**Comment by rbalykov** [comment]
Framerate:

It could be 44,13 Hz as of 1989 papers.
Using new rates, it could be 1412,18 Hz using single universe at 8M.

Brought to TV's 60 Hz, it gives 23,53 Hz.
So transmitting 20U (universes) to cable, 3U could be received back by console to take some RDM.

Brought to ws2812's 500 Hz internal refresh, it gives 2,82. In other words, 18 meters of WS pixels 60led/m rgb could be refreshed at 500 Hz with same XLR cable while driving 32U.

**Comment by rbalykov** [comment]
5pin XLR rework

Half-duplex cables remain unchainged.
Fullduplex cables give RS-422 topology, hardly needed by stage kinematic control. Requies new splitter/ip-gate that could be made with OLA and RPi.

Modem baudrates

Thats more heritage to adopt.
Using up to 24V peak-to-peak pulse as far in 1970's, allows your old i386 to drive DMX right from DIN at 9600n2 rate to achieve 1,69 DMX frames per sec.
